examples/apps/color_finder.py

```diff
@@ -0,0 +1,710 @@
+        self.blobs_dict[left_blob_key].append((i, j))
+        self.keys.at(i, j).set(left_blob_key)
+        for (x, y) in above_blob_points:
+            self.keys.at(x, y).set(left_blob_key)
+        self.blobs_dict.pop(above_blob_key)
+
+    def filter_blobs_dict_by_size(self, n):
+        '''Filters out small blobs from self.blobs_dict.
+
+            Args:
+                n (int): the number of points required of a blob to stay in self.blobs_dict
+
+        '''
+        self.blobs_dict = dict((blob, list_of_points) for blob, list_of_points in self.blobs_dict.items() if len(list_of_points) >= n)
+
+    def get_largest_blob_with_color(self, color):
+        '''Finds the largest blob of a particular color
+
+            Args:
+                color (string): the desired color of the blob
+
+            Returns:
+                the key of the largest blob with that color, or None if no such blob exists
+
+        '''
+        filtered_dict = dict(((n, k), v) for (n, k), v in self.blobs_dict.items() if k == color)
+        values = filtered_dict.values()
+        if len(values) > 0:
+            longest_blob_list = functools.reduce(lambda largest, current: largest if (largest > current) else current, values)
+            sample_x, sample_y = longest_blob_list[0]
+            return self.keys.at(sample_x, sample_y).value
+        else:
+            return None
+
+    def make_blob_centers(self):
+        '''Constructs a dictionary to keep track of the center of each blob.
+
+            Returns:
+                info (dict) : the dictionary of blob centers
+                    Key: color (string): the color of the blob
+                    Value: x, y (int, int): the approximate center of the blob as coordinates of self.matrix
+
+        '''
+        info = {}
+        biggest_blobs = []
+        for color in self.keylist:
+            biggest_blobs.append(self.get_largest_blob_with_color(color))
+        for blob in biggest_blobs:
+            if blob:
+                num, color = blob
+                xs = []
+                ys = []
+                for (x, y) in self.blobs_dict[blob]:
+                    xs.append(x)
+                    ys.append(y)
+                average_x = functools.reduce((lambda a, b : a+b), xs)/len(xs)
+                average_y = functools.reduce((lambda a, b : a+b), ys)/len(ys)
+                info[color] = (int(average_x), int(average_y))
+        return info
+
+    def get_center_of_blob_with_color(self, color):
+        '''Returns the center of a blob if there is a blob with the specified color.
+        Otherwise, returns None.
+
+        Args:
+            color (string): the desired color of a blob
+
+        Returns:
+            d[color] ((int, int)): the location of the center of the desired blob, 
+            or None if there is no blob of that color
+
+        '''
+        d = self.make_blob_centers()
+        if color in d:
+            return d[color]
+        return None
+
+
+class MyMatrix():
```

Comment:
NON-IMMEDIATELY-ACTIONABLE:
I wouldn't bother doing it with this PR, but I would be interested in seeing something like this merged into our SDK, with some support for writing to and from images and numpy arrays.

I ran into a very similar situation on my "outline shader" program, in that neighborhood context is useful in low level image processing and the numpy library doesn't really play well with running functions that deal with more than just one specific member.